packages/knip/src/plugins/vitest/index.ts

```diff
@@ -84,6 +112,38 @@ export const resolveConfig: ResolveConfig<ViteConfigOrFn | VitestWorkspaceConfig
 
   const configs = await getConfigs(localConfig);
 
+  // Handle external project files from glob patterns
+  const externalProjectFiles: string[] = [];
+  for (const cfg of configs) {
+    if (cfg.test?.projects) {
+      for (const project of cfg.test.projects) {
+        if (typeof project === 'string') {
+          // Resolve glob patterns to actual files
+          const projectFiles = await _glob({
+            cwd: options.cwd,
+            patterns: [project],
+            gitignore: false,
+          });
+          externalProjectFiles.push(...projectFiles);
```

Comment:
Knip plugins can use `toConfig` to reference a config file.

Unfortunately this function doesn't support a specifier with a glob pattern, so indeed we do need to `_glob` them separately as we have it already. But the nice part of adding the `toConfig()` input is that we should no longer need to `_load()` the file in the plugin: `resolveConfig` will be executed again with that file's config.

This should also result in the dependencies being "assigned" to the workspace where the other config file is.